### ISSUE TITLE
Gblame navigation

### ISSUE THREAD

**Issue by tkhai** [tpope/vim-fugitive#1422]
This feature request is about possibility to navigate blame commits using a single key instead of keys combination. Say, we have:

commit1 (Author1              Date1)
commit2 (Author2              Date2)

and we want to see the both of commits sequentially.
Currently, this is possible only via opening in a new tab:
1)Press 'O' on commit1 (i.e., 'Shift + o' -- 2 keys)
2)Press ":q<CR> to return back (3 keys).
3)Goto commit2, etc
In case of there are a lot commits to see, this excess keys become a tiredness.
The problem is that there is no a way to remap 'Shift + o' to a single key, and the same is with ':q<CR>'. Another problem is that opening occurs in a new tab, that is not always comfortable for a user (tab is only possibility to open commit in full screen and return back later).

It would be great to have both of:
1)possibility to open commit in the same buffer (not in a new tab) via single key (configurable);
2)possibility to return back to blame lines list via a single key (configurable).

Thanks!

**Comment by odnoletkov** [comment]
<kbd>p</kbd> might be a shortcut you are looking for:
```
                                                *fugitive_p*
p                       Open the file or |fugitive-object| under the cursor in
                        a preview window.  …
```

**Comment by tpope** [comment]
> 1)possibility to open commit in the same buffer (not in a new tab) via single key (configurable);

You mean open it in the blame window? That window has a really awkward width; I don't think this is a good idea.

> 2)possibility to return back to blame lines list via a single key (configurable).

I don't provide configuration options for things like this. You can `nmap <whatever> :q<CR>` without my help. (If I did your first idea, it would be `nmap <whatever> :e #<CR>`, but I don't think I'll be doing that.)

**Comment by tkhai** [comment]
> You mean open it in the blame window? That window has a really awkward width; I don't think this is a good idea.

Yes, I mean that. It should be possible to resize blame window on commit opening.

>You can nmap

This is only possible in case of opening a commit in a new tab. But my suggestion in general is to open a commit in the blame window.

**Comment by tpope** [comment]
Resize to what width? I don't think there's a one size fits all solution here.

**Comment by tkhai** [comment]
1)It does not matter . Something like 95% of screen or (screen_width - 1).

let c = float2nr(&columns * 100 / 95)
execute "vertical resize" . c

2)I just clarify my use case. I want to have a possibility to iterate blame commits on 5-10 neighbor strings. It's needed to return from commit back to blame window. Currently, the only possibility is to open a commit in a new tab with <Shift+o>, which is not comfortable. I always map frequently-used  combinations in a single key, but for Gblame it's not possible.

3)Also, it's not very intuitive, that <CR> in blame window destroys the window without any possibility to return back. Because Enter is common key for many utils. E.g., mc never deprives you a possibility to return to parent directory after you entered a child directory. The same is vim's netrw navigation.

Sadly, but Gblame behaves in another way. I use Gblame once a week, but every time I forget, that <Enter> kills blame window, and I have to wait second Gblame call after I pressed <Enter> and jumper to wrong commit.

I even have such the comment in vimrc, just to stop press <Enter> and to save my time on Gblame second call:
nnoremap <S-F6> :Gblame<CR>:echo "O -- open commit in new tab, :q -- close commit"<CR>

So, maybe I had to write all of this in the initial description, but it's better later than never.

My experience is a user: jumping to wrong commit is a common case. Necessity to check commits of neighbour lines is also necessity. But historical-intuitive key (Enter) is not intuitive in Gblame. This is a big problem for me.

Possibility to return back after <Enter> would be solved the problem for me.
